src/arch/arm/64/idle.c

```diff
@@ -27,6 +29,7 @@ void NORETURN NO_INLINE VISIBLE halt(void)
     debug_printKernelEntryReason();
 #endif
 #endif
+    wdog_reset();
```

Comment:
If we declare `wdog_reset()` as `weak`, then we could use
```
if (wdog_reset) {
    wdog_reset(); /* this is no expected to return */
}
```
And it would also compile on platforms where this is not implemented

Reply:
That would work, but it turns out the AOS branch already doesn't compile for other platforms, so it's probably not worth it.